### PR TITLE
Wip/superm1/synaptics error handling

### PR DESCRIPTION
**Pull request by superm1**
Not really intended for holding up 1.1.1, needs to be better tested across hardware combinations (leaf/tesla direct/cascaded, panamera).  So if it slips 1.1.1 no big deal to me.

I did some light testing on leaf/tesla direct and shook out a bug already.

### CODE REVIEW

File: plugins/synapticsmst/synapticsmst-device.c [hunk 1]
```diff
@@ -8,11 +8,8 @@
  */
 
 #include "config.h"
-#include <string.h>
```
Comment (hughsie) [inline plugins/synapticsmst/synapticsmst-device.c]: I think you need this one: `error: implicit declaration of function ‘memset’`

File: plugins/synapticsmst/synapticsmst-device.c [hunk 2]
```diff
@@ -247,24 +247,26 @@ synapticsmst_device_scan_cascade_device (SynapticsMSTDevice *device,
...
 	priv->has_cascade = FALSE;
 
-	if (!synapticsmst_device_enable_remote_control (device, error)) {
-		g_prefix_error (error,
-				"failed to enable remote control on tx_port %d: ",
-				tx_port);
+	/* enable remote control and disable on exit */
+	locker = fu_device_locker_new_full (device,
+					    (FuDeviceLockerFunc) synapticsmst_device_enable_remote_control,
+					    (FuDeviceLockerFunc) synapticsmst_device_disable_remote_control,
```
Comment (hughsie) [inline plugins/synapticsmst/synapticsmst-device.c]: I think this is exactly how you should use FuDeviceLocker -- much appreciated. Why not just call it `synapticsmst_device_open()` and `synapticsmst_device_close()`? Then it's very obvious what it's doing.

**Comment by hughsie** [comment]
The `GError` 2nd commit is most welcome to see. :)

File: plugins/synapticsmst/synapticsmst-device.c (hunk 2 above)
Comment (superm1) [reply plugins/synapticsmst/synapticsmst-device.c]: Actually it's not needed for all communication requests, only specific ones.  These commits are the basis to allow code reuse across plugins (symlink the source files and compile into both plugins) and those may not necessarily use remote control functionality (TBD).

Comment (hughsie) [reply plugins/synapticsmst/synapticsmst-device.c]: Ahh okay, NP then.

**Comment by hughsie** [comment]
LGTM.

**Pull request merged**